### DISCUSSION TITLE
Tracing | Automatic Tracing for Assistants API

### DISCUSSION THREAD

**Discussion by hayescode** [microsoft/promptflow#3347] [Ideas]
I really like the tracing package so far! I use Azure OpenAI, Assistants API, and streaming.

My dream would be toggle certain model deployments for tracing, so that all API calls to an Azure OpenAI endpoint/deployment will be traced for usage later.

This would greatly simplify development and increase usage of Azure AI Studio.

**Comment by zhengfeiwang** [comment]
@luigiw could you please share any update on tracing for Assistants API?

**Comment by M-Hietala** [comment]
Tracing of Assistants API in PromptFlow is something we are currently working on.

**Reply by hayescode** [reply]
@M-Hietala any ETA for Assistant tracing (and usage in Promptflow LLM tool)?

**Comment by anderl80** [comment]
That's an interesting demo using opentelemetry to do the tracing https://github.com/microsoft/assistant-pf-demo
